dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/SwitchExpressionNode.java

```diff
@@ -0,0 +1,87 @@
+package org.checkerframework.dataflow.cfg.node;
+
+import com.sun.source.tree.Tree;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/** A node for a switch expression. */
+public class SwitchExpressionNode extends Node {
+
+  /** The switch expression tree corresponding to this node. */
```

Comment:
Does this have a specific run-time type?  If so, document that.  Probably the constructor should throw an exception if an object of the wrong type is passed in.